stdlib/LinearAlgebra/src/tridiag.jl

```diff
@@ -170,6 +170,10 @@ adjoint(S::SymTridiagonal) = Adjoint(S)
 Base.copy(S::Adjoint{<:Any,<:SymTridiagonal}) = SymTridiagonal(map(x -> copy.(adjoint.(x)), (S.parent.dv, S.parent.ev))...)
 Base.copy(S::Transpose{<:Any,<:SymTridiagonal}) = SymTridiagonal(map(x -> copy.(transpose.(x)), (S.parent.dv, S.parent.ev))...)
 
+ishermitian(S::SymTridiagonal{<:Real}) = true
+ishermitian(S::SymTridiagonal{<:Complex}) = sum(imag, S.dv) == 0 && sum(imag, S.ev) == 0
```

Comment:
You require the diagonal to be real, not just the sum of the imaginary parts to be 0. Take `[im 0; 0 -im]` for example.

Reply:
Good catch, I was trying to come up with something faster than `all(x -> imag(x) == 0, S.dv) && all(x -> imag(x) == 0, S.ev)`. Fixed now, and added a few tests.